hudi-spark-datasource/hudi-spark3/src/main/scala/org/apache/spark/sql/hudi/analysis/HoodieSpark3Analysis.scala

```diff
@@ -45,16 +45,22 @@ case class HoodieSpark3Analysis(sparkSession: SparkSession) extends Rule[Logical
   with SparkAdapterSupport with ProvidesHoodieConfig {
 
   override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsDown {
-    case dsv2 @ DataSourceV2Relation(d: HoodieInternalV2Table, _, _, _, _) =>
-      val output = dsv2.output
-      val catalogTable = if (d.catalogTable.isDefined) {
-        Some(d.v1Table)
-      } else {
-        None
-      }
-      val relation = new DefaultSource().createRelation(new SQLContext(sparkSession),
-        buildHoodieConfig(d.hoodieCatalogTable))
-      LogicalRelation(relation, output, catalogTable, isStreaming = false)
+    // NOTE: This step is required since Hudi relations don't currently implement DS V2 Read API
+    case dsv2 @ DataSourceV2Relation(tbl: HoodieInternalV2Table, _, _, _, _) =>
+      val qualifiedTableName = QualifiedTableName(tbl.v1Table.database, tbl.v1Table.identifier.table)
+      val catalog = sparkSession.sessionState.catalog
+
+      catalog.getCachedPlan(qualifiedTableName, () => {
```

Comment:
So the issue is that this cache is never invalidated, on write. and V1 does not have a notion of Catalog (to be used for writes)

Reply:
@leesf I propose we revert back to v1 and push out 0.11.1 to fix these perf regressions. Do you see any concerns with that ? 

@YannByron would any of the follow-on sql work break if we revert?

Reply:
no. v1 and v1 are just different in the internal workings of spark, and having a few more configurations. The interface to the user is not actually affected.

Reply:
> 

@vinothchandar revert back to v1 means drop `HoodieCatalog`? If so, I do not think it is a good idea since users use 0.11.0 would use `HoodieCatalog` config and will change the behavior if we drop the `HoodieCatalog` which is not a good idea.

Reply:
I am just asking for ideas to fix this.